projection_utils.py

```diff
@@ -0,0 +1,324 @@
+    """
+    f = open(axisfile)
+    axis_align_matrix = np.array(
+        f.readline().strip().split(" ")[
+            2:]).astype(np.float32)
+    axis_align_matrix = np.reshape(axis_align_matrix, (4, 4))
+
+    return axis_align_matrix
+
+
+def read_mesh_vertices(filename):
+    """ read XYZ for each vertex.
+    """
+    assert os.path.isfile(filename)
+    with open(filename, 'rb') as f:
+        plydata = PlyData.read(f)
+        num_verts = plydata['vertex'].count
+        vertices = np.zeros(shape=[num_verts, 3], dtype=np.float32)
+        vertices[:, 0] = plydata['vertex'].data['x']
+        vertices[:, 1] = plydata['vertex'].data['y']
+        vertices[:, 2] = plydata['vertex'].data['z']
+    return vertices
+
+
+def alignaxis(axisfile, plyfile, sample=False):
+    """Align the point clouds using the axis alignment matrix
+        Args:
+          axisfile: file containig the axis alignment matrix
+          plyfile: point cloud file
+
+        Return: x,y,z numpy array of 5000 points if sample is true
+    """
+    assert os.path.isfile(axisfile)
+    axis_align_matrix = get_axismatrix(axisfile)
+    mesh_vertices = read_mesh_vertices(plyfile)
+
+    # align the vertices to axis co-ordinate
+    pts = np.ones((mesh_vertices.shape[0], 4))
+    pts[:, 0:3] = mesh_vertices[:, 0:3]
+    # rigid body transformation of the points
+    pts = np.matmul(axis_align_matrix, pts.transpose()).T
+    mesh_vertices[:, 0:3] = pts[:, 0:3]
+
+    if sample:
+        pts = np.arange(pts.shape[0])
+        pts = np.random.choice(pts, 5000, replace=False)
+
+    return mesh_vertices[pts]
+
+
+def rotationmat(heading_angle):
+    """Rotate axis by heading angle
+
+        heading_angle: angle of rotation
+        Return: 3x3 rotation matrix
+    """
+    rotmat = np.zeros((3, 3))
+    rotmat[2, 2] = 1
+    cosval = np.cos(heading_angle)
+    sinval = np.sin(heading_angle)
+    rotmat[0:2, 0:2] = np.array([[cosval, -sinval], [sinval, cosval]])
+    return rotmat
+
+
+def local_coord(scale):
+    """Define a 3d coord cube centered at 0
+
+        scale: size 3 list/array for size of the cube
+        Return: 9x3 array of local vertices for cube with center at 0
+    """
+    w = scale[0] / 2.
+    h = scale[1] / 2.
+    d = scale[2] / 2.
+
+    # Define the local coordinate system, w.r.t. the center of the box
+    box_coords = np.array([[0., 0., 0.], [-w, -h, -d], [-w, -h, +d], [-w, +h, -d],
+                           [-w, +h, +d], [+w, -h, -d], [+w, -h, +d], [+w, +h, -d],
+                           [+w, +h, +d]])
+    return box_coords
+
+
+def gen_vertices(box_points, heading_angle):
+    """Generate homogenous vertices of bounding boxes given center, size and heading angle of box
+        Args:
+          boxpoint: size 7 array containing box center, size and angle in degrees
+
+        Return: 9x4 array of vertices for bounding box
+    """
+    # convert angle to radians
+    heading_angle = heading_angle * np.pi / 180
+
+    ctr = box_points[:3]
```

Comment:
Use descriptive names when not short on space, like center, transformations, etc. makes reading your code a lot easier.